.azure-devops/build/steps/macOS/pre.yml

```diff
@@ -0,0 +1,76 @@
+# pre.yml
+# install macOS platform dependencies & set environment variables
+
+parameters:
+  # store all the downloaded files
+  dependenciesDir: "$(Agent.BuildDirectory)/tmp"
+  # directory to download the JDK used to bootstrap the build
+  jdkBootDir: "$(Agent.BuildDirectory)/jdk/boot"
+  # store jdk used for gradle
+  javaHomeDir: "$(Agent.BuildDirectory)/jdk/home"
+
+steps:
+  # based on
+  # https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/master/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/scripts/install-xcode.sh
+  - bash: |
+      touch /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress;
+      PROD=$(softwareupdate -l |
+        grep "\*.*Command Line" |
+        head -n 1 | awk -F"*" '{print $2}' |
+        sed -e 's/^ *//' |
+        tr -d '\n')
+      softwareupdate -i "$PROD" --verbose;
+      rm /tmp/.com.apple.dt.CommandLineTools.installondemand.in-progress;
+    displayName: "[macOS Pre] install Xcode command line tools"
+
+  # based on
+  # https://github.com/AdoptOpenJDK/openjdk-infrastructure/blob/master/ansible/playbooks/AdoptOpenJDK_Unix_Playbook/roles/Common/vars/MacOSX.yml
+  - bash: |
+      brew install autoconf ccache coreutils wget
+    displayName: "[macOS Pre] brew install autoconf, ccache, coreutils, and wget"
+
+  # cache task to cache dependencies, e.g, jdk8 and bootjdk
+  - task: Cache@2
+    inputs:
+      key: '"$(Agent.OS)"|$(Build.SourcesDirectory)/.azure-devops/build/steps/macOS/pre.yml'
+      path: "${{ parameters.dependenciesDir }}"
+      cacheHitVar: CACHE_RESTORED
+    displayName: "[macOS Pre] caching download dependencies"
+
+  # create all required directories before downloading files & installing jdks
+  - bash: |
+      mkdir -p ${{ parameters.jdkBootDir }}
+      mkdir -p ${{ parameters.javaHomeDir }}
+      mkdir -p ${{ parameters.dependenciesDir }}
+    displayName: "[macOS Pre] create required directories"
+
+  # download JDKs from AdoptOpenJDK
+  - bash: |
```

Comment:
What about for Java 8?  Need to get a Java 7 boot.  Adopt doesn't have these, so I recommend Zulu